### PR TITLE
Mpdp 404 search suggestions

### PR DESCRIPTION
**Pull request by vmuthu-kn**

### CODE REVIEW

File: app/services/fuzzySearchService.js [hunk 1]
```diff
@@ -102,6 +102,13 @@ const groupByPayee = (searchResults) => {
   return result
 }
 
+const getSearchSuggestions = async (searchKey) => {
+  const paymentData = await getAllPaymentData()
+  const fuse = new Fuse(paymentData, fuseSearchOptions)
+  const searchResult = fuse.search(searchKey).map(row => row.item)
+  return searchResult.slice(0, 6)
```
Comment (SidG-CG) [inline app/services/fuzzySearchService.js]: Can we define the magic number 6 somewhere as the search suggestion results limit?

File: app/services/fuzzySearchService.js [hunk 2]
```diff
@@ -102,6 +102,13 @@ const groupByPayee = (searchResults) => {
   return result
 }
 
+const getSearchSuggestions = async (searchKey) => {
+  const paymentData = await getAllPaymentData()
+  const fuse = new Fuse(paymentData, fuseSearchOptions)
+  const searchResult = fuse.search(searchKey).map(row => row.item)
```
Comment (SidG-CG) [inline app/services/fuzzySearchService.js:111]: Have you tested this? I think just returning results from search will end up returning same payee twice if that payee has received payment for both schemes.

Comment (SidG-CG) [reply app/services/fuzzySearchService.js:111]: If you look at the getPaymentData function, we call groupByPayee function to get bundle same payees together

File: app/services/fuzzySearchService.js (hunk 1 above)
Comment (vmuthu-kn) [reply app/services/fuzzySearchService.js]: agreed, and moved to appconfig file now.

File: app/services/fuzzySearchService.js (hunk 2 above)
Comment (vmuthu-kn) [reply app/services/fuzzySearchService.js:111]: good point, the groupby is applied now.

File: app/services/fuzzySearchService.js (hunk 1 above)
Comment (SidG-CG) [reply app/services/fuzzySearchService.js]: I think we can simplify searchSuggestionResultsLimit if we put it within the search object in appConfig and the name can also be shortened to suggestionResultsLimit as we know its the search config.

Comment (vmuthu-kn) [reply app/services/fuzzySearchService.js]: ok, moved the variable & formatted the name now.

**Review by SidG-CG — APPROVED** [review]

**Pull request merged**